src/iosrv/init.c

```diff
@@ -589,10 +590,57 @@ server_init(int argc, char *argv[])
 	return rc;
 }
 
+static int
+cleanup_cb(const char *path, const struct stat *sb, int flag,
+	   struct FTW *ftwbuf)
+{
+	int rc;
+
+	if (ftwbuf->level == 0)
+		return 0;
+
+	if (flag == FTW_DP || flag == FTW_D)
+		rc = rmdir(path);
+	else
+		rc = unlink(path);
+	if (rc)
+		D_ERROR("failed to remove %s\n", path);
+	return rc;
+}
+
+static void
+shared_memory_file_cleanup(void)
+{
+	int rc;
+
+	if (dss_nvme_shm_id != DAOS_NVME_SHMID_NONE) {
```

Comment:
I was under the impression that we are now not using shared memory identifiers and not running in multiprocess mode, so if that's the case then will this ever be executed? @mjmac @jbrosenz

Reply:
We are indeed setting a shmid per ioserver: https://github.com/daos-stack/daos/blob/master/src/control/server/server.go#L165

The difference from before is that the ioservers are now primary processes and daos_admin will need to connect as a secondary in order to interact with devices held by the ioservers.